src/OmniSharp.Host/Services/SharedTextWriter.cs

```diff
@@ -19,26 +19,33 @@ public SharedTextWriter(TextWriter writer)
         protected override void DisposeCore(bool disposing)
         {
             // Finish sending queued output to the writer.
-            _task.Wait();
+            _semaphoreSlim.WaitAsync().GetAwaiter().GetResult();
         }
 
         public void WriteLine(object value)
```

Comment:
Can we refactor this method away?

Reply:
looking at the code it looks like it's called in other async contexts.

Reply:
Yeah, it's not used anymore. As we don't ship NuGets, we probably don't care about obsoleting, just remove it from the interface?

Reply:
yeah, I say just nuke it before we start shipping nugets 😄 👍